### PR TITLE
✨ Source Metabase: Add API Key Authentication

### LINKED ISSUE #45356: [source-metabase] Add API Key Auth
### Connector Name

source-metabase

### Connector Version

2.1.1

### What step the error happened?

None

### Relevant information

The preferred way of authenticating to the [Metabase API is via API Keys](https://www.metabase.com/docs/latest/people-and-groups/api-keys) over the current implemented way (user login/ session token)

### Relevant log output

_No response_

### Contribute

- [X] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by kempspo**
## What
Resolves https://github.com/airbytehq/airbyte/issues/45356

The preferred way of authenticating to the [Metabase API is via API key](https://www.metabase.com/docs/latest/people-and-groups/api-keys) over the current implemented way (user login/ session token).

## How
Adds `SelectiveAuthenticator` to allow users to select between session auth and API key auth.

Note: this PR also cleans up the `manifest.yaml`, I can split this out into a different PR if needed.

## Review guide
1. `manifest.yaml`

## User Impact
On source creation, users can now select between session auth and API key auth.

## Can this PR be safely reverted and rolled back?
- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: docs/integrations/sources/metabase.md [hunk 1]
```diff
@@ -77,6 +97,7 @@ The Metabase source connector supports the following [sync modes](https://docs.a
 
 | Version | Date       | Pull Request                                             | Subject                                                                                                                          |
 | :------ | :--------- | :------------------------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------- |
+| 2.2.0   | 2024-08-27 |
```
Comment (marcosmarxm) [inline docs/integrations/sources/metabase.md:100]: This is a breaking change as old connections will need to update their config in the UI please bump accordly.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Left one comment.

File: docs/integrations/sources/metabase.md (hunk 1 above)
Comment (kempspo) [reply docs/integrations/sources/metabase.md:100]: This makes sense. I actually want an opinion on another thing.

Currently, Metabase source uses the `LegacySessionTokenAuthenticator`, my PR updates it to use the new `SessionTokenAuthenticator`. Do we prefer to keep the old one or should we use the new one?

Also, I realized I didn't bring it up here in the PR but in [Slack](https://airbytehq.slack.com/archives/C054V9JFTC6/p1725177930769979), I was having trouble with `SelectiveAuthenticator` to pass down some properties to the `SessionTokenAuthenticator`.

**Comment by georgek** [comment]
Can this be re-opened?

**Pull request closed without merging**